Installation issue: shtools

### Steps to reproduce the issue

```console
spack install shtools
```

### Information on your system

```
* **Spack:** 0.16.3-4915-1d1beb7
* **Python:** 3.6.8
* **Platform:** linux-centos7-broadwell
* **Concretizer:** clingo
```

### Additional information


[spack-build-out.txt](https://github.com/spack/spack/files/7428897/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/7428898/spack-build-env.txt)


The problem seems to be related to libtool, even though it's installed correctly as dependency
```
==> Error: ProcessError: Command exited with status 2:
    'make' '-j10' 'fortran' 'F95=/att/gpfsfs/briskfs01/ppl/sberton2/.spack_repo/opt/spack/linux-centos7-broadwell/gcc-11.1.0/gcc-9.3.0-edodhooi2frihm46komjn2blyd6nmatk/bin/gfortran' 'F95FLAGS=-fPIC -O3 -std=gnu -ffast-math' 'OPENMPFLAGS=-fopenmp' 'BLAS=/att/gpfsfs/briskfs01/ppl/sberton2/.spack_repo/opt/spack/linux-centos7-broadwell/gcc-9.3.0/openblas-0.3.18-3n6bxikyxyv7plf7fljswyckps73i5de/lib/libopenblas.so' 'FFTW=/att/gpfsfs/briskfs01/ppl/sberton2/.spack_repo/opt/spack/linux-centos7-broadwell/gcc-9.3.0/fftw-3.3.9-bxa3x3hvm2sgoyr4wptbl2s7fljtmgqp/lib/libfftw3.so' 'LAPACK=/att/gpfsfs/briskfs01/ppl/sberton2/.spack_repo/opt/spack/linux-centos7-broadwell/gcc-9.3.0/openblas-0.3.18-3n6bxikyxyv7plf7fljswyckps73i5de/lib/libopenblas.so' 'PREFIX=/att/gpfsfs/briskfs01/ppl/sberton2/.spack_repo/opt/spack/linux-centos7-broadwell/gcc-9.3.0/shtools-4.8-3uji7smuqkz5wxgf2x7alaii6onev5zw' 'PWD=/att/nobackup/sberton2/.tmp/spack-stage/spack-stage-shtools-4.8-3uji7smuqkz5wxgf2x7alaii6onev5zw/spack-src'

2 errors found in build log:
     118    /att/gpfsfs/briskfs01/ppl/sberton2/.spack_repo/opt/spack/linux-centos7-broadwell/gcc-11.1.0/gcc-9.3.0-edodhooi2frihm46komjn2blyd6nmatk/bin/gfortran -c -fPIC -O3 -std=gnu -ffast-math MakeMagGradGridD
            H.f95
     119    
     120    --> Compilation of source files successful
     121    
     122    libtool -static -o /att/nobackup/sberton2/.tmp/spack-stage/spack-stage-shtools-4.8-3uji7smuqkz5wxgf2x7alaii6onev5zw/spack-src/lib/libSHTOOLS.a ftypes.o SHTOOLS.o FFTW3.o PlanetsConstants.o CilmPlus.
            o CilmMinus.o ComputeDG82.o ComputeDm.o DHaj.o djpi2.o BAtoHilm.o MakeGrid2D.o GLQGridCoord.o MakeGridPoint.o MakeGridPointC.o PlBar.o PlBar_d1.o PLegendre.o PLegendre_d1.o PLegendreA.o PLegendreA_d
            1.o PlmBar.o PlmBar_d1.o PlmIndex.o PlmSchmidt.o PlmSchmidt_d1.o PlSchmidt.o PlSchmidt_d1.o PreGLQ.o Random.o SHAdmitCorr.o SHBias.o SHBiasK.o SHConvertCoef.o SHFindLWin.o SHGLQ.o SHLocalizedAdmitCo
            rr.o SHMultiply.o SHMultiTaperCSE.o SHMultiTaperSE.o SHPowerSpectra.o SHRead.o SHReadJPL.o SHReturnTapersM.o SHRotateCoef.o SHRotateRealCoef.o SphericalCapCoef.o Wigner3j.o DownContFilter.o SHRead2.
            o MakeGeoidGrid.o MakeCircleCoord.o SHMTCouplingMatrix.o SHReturnTapers.o SHSjkPG.o PlON.o PlON_d1.o PlmON.o PlmON_d1.o SHConfidence.o SHMagPowerSpectra.o SHPowerSpectraC.o SHBiasAdmitCorr.o SHCilmT
            oVector.o YilmIndexVector.o ComputeDMap.o SHReturnTapersMap.o Curve2Mask.o MakeEllipseCoord.o CilmPlusRhoH.o CilmMinusRhoH.o BAtoHilmRhoH.o NormalGravity.o SHMultiTaperMaskSE.o SHMultiTaperMaskCSE.o
             SHBiasKMask.o SHRotateTapers.o SlepianCoeffs.o SlepianCoeffsToSH.o SHSCouplingMatrix.o SHMTVar.o SHSlepianVar.o SHSCouplingMatrixCap.o cWrapper.o MakeGravGridPoint.o MakeMagGridPoint.o MakeGradient
            DH.o EigValSym.o EigValVecSym.o EigValVecSymTri.o SHExpandLSQ.o SHMTDebias.o SHMTVarOpt.o MakeGridDH.o MakeGridDHC.o MakeGridGLQ.o MakeGridGLQC.o SHExpandDH.o SHExpandDHC.o SHExpandGLQ.o SHExpandGLQ
            C.o MakeGravGradGridDH.o MakeGravGridDH.o MakeMagGridDH.o MakeMagGradGridDH.o
     123    make[1]: libtool: Command not found
  >> 124    make[1]: *** [/att/nobackup/sberton2/.tmp/spack-stage/spack-stage-shtools-4.8-3uji7smuqkz5wxgf2x7alaii6onev5zw/spack-src/lib/libSHTOOLS.a] Error 127
     125    make[1]: Leaving directory `/att/gpfsfs/briskfs01/ppl/sberton2/.tmp/spack-stage/spack-stage-shtools-4.8-3uji7smuqkz5wxgf2x7alaii6onev5zw/spack-src/src'
  >> 126    make: *** [fortran] Error 2
```
I also tried different compilers (gcc@11.1.0) and versions (shtools@4.5 also ends with an error, though with a different one ). It would be great if @eschnett could have a look at this. Thanks!

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/shtools/nolibtool.patch
+++ b/var/spack/repos/builtin/packages/shtools/nolibtool.patch
@@ -1,0 +1,16 @@
+--- a/src/Makefile
++++ b/src/Makefile
+@@ -80,10 +80,10 @@
+ 	@echo "--> Compilation of source files successful"
+ 	@echo
+ 	@rm -f $(PROG)
+-	$(LIBTOOL) $(LIBTOOLFLAGS) -o $(PROG) $(OBJS)
++#	$(LIBTOOL) $(LIBTOOLFLAGS) -o $(PROG) $(OBJS)
+ #	If you prefer to use libtool, uncomment the above line, and comment the two lines below (AR and RLIB)
+-#	$(AR) $(ARFLAGS) $(PROG) $(OBJS)
+-#	$(RLIB) $(RLIBFLAGS) $(PROG)
++	$(AR) $(ARFLAGS) $(PROG) $(OBJS)
++	$(RLIB) $(RLIBFLAGS) $(PROG)
+ 	@echo
+ 	@echo "--> Creation of static library successful"
+ #	@rm -f $(OBJS)

--- a/var/spack/repos/builtin/packages/shtools/package.py
+++ b/var/spack/repos/builtin/packages/shtools/package.py
@@ -31,6 +31,11 @@ class Shtools(MakefilePackage):
     depends_on('blas')
     depends_on('fftw')
     depends_on('lapack')
+    depends_on('py-flake8', type='test')
+
+    def patch(self):
+        """make check fix: Silence "do not use bare 'except'" in number of files"""
+        filter_file('ignore=', 'ignore=E722,', 'Makefile')
 
     # Options for the Makefile
     def makeopts(self, spec, prefix):

--- a/var/spack/repos/builtin/packages/shtools/package.py
+++ b/var/spack/repos/builtin/packages/shtools/package.py
@@ -14,12 +14,17 @@ class Shtools(MakefilePackage):
 
     maintainers = ['eschnett']
 
+    version('4.9.1', sha256='5c22064f9daf6e9aa08cace182146993aa6b25a6ea593d92572c59f4013d53c2')
     version('4.8', sha256='c36fc86810017e544abbfb12f8ddf6f101a1ac8b89856a76d7d9801ffc8dac44')
     version('4.5', sha256='1975a2a2bcef8c527d321be08c13c2bc479e0d6b81c468a3203f95df59be4f89')
 
     # Note: This package also provides Python wrappers. We do not
     # install these properly yet, only the Fortran library is
     # installed.
+
+    # The Makefile expects the "other" libtool, not the GNU libtool we have in
+    # Spack
+    patch('nolibtool.patch')
 
     variant('openmp', default=True, description="Enable OpenMP support")
 

